connectors/tests/test_utils.py

```diff
@@ -164,3 +168,72 @@ def test_get_base64_value():
     """This test verify get_base64_value method and convert encoded data into base64"""
     expected_result = get_base64_value("dummy".encode("utf-8"))
     assert expected_result == "ZHVtbXk="
+
+
+@pytest.mark.asyncio
+async def test_concurrent_runner(patch_logger):
+    results = []
+
+    def _cb(result):
+        results.append(result)
+
+    async def coro(i):
+        await asyncio.sleep(0.1)
+        return i
+
+    runner = ConcurrentTasks(results_callback=_cb)
+    for i in range(10):
+        await runner.put(functools.partial(coro, i))
+
+    await runner.wait()
+    assert results == list(range(10))
+
+
+@pytest.mark.asyncio
+async def test_concurrent_runner_fails(patch_logger):
+    results = []
+
+    def _cb(result):
+        results.append(result)
+
+    async def coro(i):
+        await asyncio.sleep(0.1)
+        if i == 5:
+            raise Exception("I FAILED")
+        return i
+
+    runner = ConcurrentTasks(results_callback=_cb)
+    for i in range(10):
+        await runner.put(functools.partial(coro, i))
+
+    with pytest.raises(Exception):
+        await runner.wait()
+
+
+@pytest.mark.asyncio
+async def test_concurrent_runner_high_concurrency(patch_logger):
+    results = []
+
+    def _cb(result):
+        results.append(result)
+
+    async def coro(i):
+        await asyncio.sleep(0)
+        return i
+
+    s_results = []
```

Comment:
Another case where naming could be clearer. `second_results` or `results2` maybe?